ScilifelabDataCentre/swedgene
# Test fixtures for GWAS plugin

We have gotten confirmation that we will soon have new population genomics data incoming. I took this as encouragement to continue the investigation of the GWAS plugin started by @Kwentine in [EB-56](https://scilifelab.atlassian.net/browse/EB-56?atlOrigin=eyJpIjoiN2UwY2U1MWJiZmVlNDc5MThmNDI4MzRmNzFiNjk2YTgiLCJwIjoiaiJ9). Big thanks for the previous progress on this! I mostly just tweaked the config and prepared new test data.

EDIT: one week later, the data has now been made public. Findings from tests of a representative file from the dataset are presented in the next post below.

--- 
The initial commit include fixtures with BED-like data and an updated version of the EB-56 config. The result is a working Manhattan plot. 

The dummy data is based on the structure of an incoming dataset I got to preview. Since that data is yet to be made public, the fixture data contains a randomized score column with a distribution based on the min-max values of the original file with some outliers sprinkled in for effect. 

To test (inspired by PR37):
```
./scripts/dockermake --test SPECIES=gwas_testing_ground
./scripts/dockerserve --dev
cp tests/fixtures/gwas_testing_ground/gwas_randomized_dummy_data.txt.bgz* hugo/static/data
cp -r tests/data/gwas_testing_ground hugo/static/data
```
And then open: [http://localhost:8080/genome-browser/?config=/data/gwas_testing_ground/config.json](http://localhost:8080/genome-browser/?config=/data/gwas_testing_ground/config.json)

Some observations:

- First I attempted to load the plain text file using `bedAdapter`, but this only resulted in an empty plot, which is similar to the result Quentin previously got with the other test data.

- Switched to `BedTabixAdapter` like in the [example in the GWAS plugin repo](https://github.com/cmdcolin/jbrowse-plugin-gwas?tab=readme-ov-file#example-config). Realized/remembered that tabix only accepts tab-delimited data ("What's in a name?" :smile: ). Reformatted with `awk '{$1=$1}1' OFS='\t' data.txt > data.tab_delim.txt`. BED sort, bgzip, tabix. The data now displayed! 

- Despite that the official specification for the BED format states that it is whitespace-delimited, tab-delimited BEDs are commonly used and are likely the easiest way to ensure compatibility with JBrowse 2. From now on, we will ask submitting researchers to tab-delimit their BED files

- BED-like files can have headers: first line need to start with # and be tab-separated. The column titles can be used to define e.g. the `scoreColumn` for the plot.

- Take-home message: ask researchers to tab-delimit their BED files.

Some things left to consider:

- Can BED-like files be used for other visualizations? Manhattan plots will not be suitable for all population genomics datasets. Histograms have already been requested.

- Should the GWAS plugin be installed in the repo instead of called remotely?

- Will the  `bedAdapter` config work with the tab-delimited data?

- The FASTA has a file size of 2 Mb despite being truncated. The config can be altered to call on the remote data instead, but a refNameAlias first need to be prepared. 

I put this a a draft PR for now because I anticipate that the incoming dataset might have additional data types that need to be tested.
